chatops-lark/pkg/events/handler/root.go

```diff
@@ -104,82 +115,260 @@ func NewRootForMessage(respondCli *lark.Client, cfg map[string]any) func(ctx con
 	cacheCfg := bigcache.DefaultConfig(10 * time.Minute)
 	cacheCfg.Logger = &log.Logger
 	cache, _ := bigcache.New(context.Background(), cacheCfg)
-	h := &rootHandler{Client: respondCli, Config: cfg, eventCache: cache}
+
+	h := &rootHandler{
+		Client:     respondCli,
+		Config:     cfg,
+		eventCache: cache,
+		logger:     log.Logger,
+	}
+
+	// Synchronously fetch bot information during initialization
+	if err := h.fetchBotInfo(context.Background()); err != nil {
+		h.logger.Fatal().Err(err).Msg("failed to fetch bot info during initialization, exiting")
+		os.Exit(1)
+	}
+
 	return h.Handle
 }
 
+// fetchBotInfo retrieves the bot information and sets the botName
+func (r *rootHandler) fetchBotInfo(ctx context.Context) error {
+	maxRetries := 3
+	retryInterval := 2 * time.Second
+
+	for i := 0; i < maxRetries; i++ {
+		if i > 0 {
+			r.logger.Info().Int("retry", i).Msg("retrying to fetch bot info")
+			time.Sleep(retryInterval)
+		}
+		// Get bot name for detect mention bot in group chat
+		// Note: In Lark API, @bot will be automatically converted to @_user_X format in message content
+		if botName, ok := r.Config["bot_name"].(string); ok && botName != "" {
+			r.botName = botName
+			r.logger.Info().Str("botName", r.botName).Msg("successfully set bot name which is from API")
+			return nil
+		}
+	}
+
+	return fmt.Errorf("failed to get bot name: bot_name not found in API")
+}
+
```

Comment:
1. maybe there's a better name for the method, such as 'initOnce'. make it will be initialized only one time.
2. the param `ctx` is not used.